Camera zoom

jennej opened the pull request:
I also would like to rename 

CameraMarkerTracker::markerToUpdate -> trackedMarker
CameraMarkerTracker::activeMarker -> selectedMarker

but held off to maybe do this after we resolve?

jennej committed:
making toString prettier

jennej committed:
right click also selects

jennej committed:
remove unnecessary argument

jennej committed:
add argument to animate function

because why do perspective cameras get to have all the fun?

jennej committed:
improved logic for layout camera centering

jennej committed:
animate all the camera transitions

jennej committed:
Merge branch 'TheAliceProject:develop' into camera_zoom

dabeshouse reviewed (changes requested):
A few things to consider.
- The zoom when focused may be larger than allowed. This means if the user tries the camera zoom arrows it can snap down to much closer. We could bump that max up, or change the logic to not snap back.
- Right click as selection is not quite there on the Mac and its weird mices. Using ctrl-click does select the thing, but the menu that pops up still applies to the last selected item. Using the two finger tap on a trackpad brings up the menu for the last selected but does not select the new item at all.
Otherwise looks good.
If you want to make the rename changes you suggest, they sound good to me, either now or later.

jennej committed:
better naming of variables

markerToUpdate -> trackedMarker
activeMarker -> selectedMarker

jennej committed:
Merge remote-tracking branch 'origin/camera_zoom' into camera_zoom

jennej committed:
handle zoom differences

a bigger max for the handles because some of our stuff is big, but within reason
controls now gracefully shove you into compliance instead of jumping back

jennej committed:
right click also selects in the pile of tiles in the left side of the scene editor
